### PR TITLE
[WIP] Documentation for rules

### PR DESCRIPTION
**Pull request by jamesqo**
This PR begins the process of adding documentation for each rule. The docs are based off the StyleCopAnalyzers documentation here: https://github.com/DotNetAnalyzers/StyleCopAnalyzers/tree/master/documentation, where each rule gets its own markdown file named after its ID. I'd like to get everyone's feedback before proceeding with further documentation, especially on the rule template.

Contributes to https://github.com/xunit/xunit/issues/1262

/cc @marcind, @ErikSchierboom

### CODE REVIEW

File: docs/xUnit2004.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+```csharp
+Assert.True(2 + 2 == 4);
+Assert.False(2 + 2 == 5);
+```
+
+## How to fix violations
+
+### For `Equal` and `StrictEqual`
+
+- `Assert.{Strict}Equal(true, b)` => `Assert.True(b)`
```
Comment (ErikSchierboom) [inline docs/xUnit2004.md:53]: I'm not sure the `{Strict}` bit makes sense here. Sure it is shorter, but I think two separate lines (one for `Assert.Equal` and one for `Assert.StrictEqual`) would be more readable.

File: docs/xUnit2004.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+- `Assert.{Strict}Equal(false, b)` => `Assert.False(b)`
+
+### For `NotEqual` and `NotStrictEqual`
+
+- `Assert.Not{Strict}Equal(true, b)` => `Assert.False(b)`
```
Comment (ErikSchierboom) [inline docs/xUnit2004.md:58]: I'm not sure the `{Strict}` bit makes sense here. Sure it is shorter, but I think two separate lines (one for `Assert.Equal` and one for `Assert.StrictEqual`) would be more readable.

**Review by ErikSchierboom — COMMENTED** [review]
Overall I think this is fantastic! The format is easy to read and very clear. Great work!

**Comment by jamesqo** [comment]
@marcind Thanks for merging this. In the future, please wait until I un-mark a PR as [WIP] before merging it. I was planning some other changes in addition to responding to PR feedback (which it looks like you took care of yourself), but was waiting for a go-ahead at https://github.com/xunit/xunit/issues/1262. I'll put those changes in a new PR.

**Pull request merged**